File: Algorithms/LinkedLists/traverse/traverse.js
/** ---------------------------------------------
 * 
 * Code to traverse a linked list
 --------------------------------------------- */


/**
 *  Class that defines our Node
 */
class node {
    constructor(value) {
        this.value = value;
        this.next  = null;
    }
}

const a = new node('A');
const b = new node('B');
const c = new node('C');
const d = new node('D');

a.next = b;
b.next = c;
c.next = d;

// A -> B -> C -> D -> NULL

/**
 * print out linked list
 */
const printLinkedList = (head) => {
    let current = head;
    
    while (current !== null) {
        process.stdout.write(current.value.toString());
        if (current.next !== null) {   
            process.stdout.write(" -> ");
        }
        current = current.next;
    }

    console.log();
};

// Call our function
printLinkedList(a);

File: Algorithms/LinkedLists/traverse/traverse.py
# ---------------------------------------------
# 
#  Code to traverse a linked list
# ---------------------------------------------

class Node:
    """ Class that defines our Node """

    def __init__(self, value):
        self.value = value
        self.next  = None

def printLinkedList(head):
    """ 
    print out linked list 

    Args:
        head (Node): first Node in Linked List.
    """

    current = head
    
    while current is not None:
        print(current.value, end="")
        if current.next is not None:
            print(" -> ", end="")
        current = current.next

if __name__ == "__main__":
    """ create a hardcoded linkedlist """

    a = Node('A')
    b = Node('B')
    c = Node('C')
    d = Node('D')

    a.next = b
    b.next = c
    c.next = d

    # call linkedlist.
    printLinkedList(a)
